### ISSUE TITLE
Reporting an already-deleted post leads to poor user experience

### ISSUE THREAD

**Issue by mcclure** [mastodon/mastodon#32314]
### Steps to reproduce the problem

Today there is some spam hitting Mastodon.
I got a piece of spam, so I reported it. When I reached the end of the "Report" flow, it turned out the moderators either on Mastodon or the originating server had already deleted the spam post. Good job! However, it turns out the UI is not as friendly as it could be when this happens.

Unfortunately, I didn't get a very good screenshot and I am not sure I can easily reproduce the circumstances on purpose. However the "report" box normally has a final "submit" button, after which it takes you to an "other actions" pane that allows you to block or mute the user. When I clicked "submit" instead I got a 404 popup in the corner, *behind* the report box, in the part of the screen that is "grayed out" when the report box is up. When I clicked multiple times, I got multiple copies of the "404" popup in the corner.

![image](https://github.com/user-attachments/assets/1cead25a-ef50-44b4-b0df-b79accf75cc1)

### Expected behaviour

The report flow should have moved forward to a pane explicitly saying the post had already been deleted and the report failed, if the user still exists (has not been banned yet), give me the opportunity to block them.

### Actual behaviour

A floaty box saying "404 record not found" in the corner

### Detailed description

Normally, I think "you tried to act on this piece of mastodon data, but it's gone" is not a very important case to optimize for. But I think this *one* case might be worth doing something about for three reasons:
- The report dialog grays out (as I remember?), and on mobile might obscure, the 404 popup.
- In the case of reporting spam, it is unusually likely that the post would be deleted by the time you are done with the report flow, more so than with other actions such as fav or boost. Therefore it is worth anticipating this case.
- Because clicking the submit button does nothing, the user is denied the chance to block the user, who might still be a problem worth blocking.

### Mastodon instance

mastodon.social

### Mastodon version

v4.3.0-rc.1+pr-32282-406f730

### Browser name and version

129.0.6668.90 (Official Build) (64-bit) (cohort: Stable) 

### Operating system

Windows 10 Version 22H2 (Build 19045.4894)

**Comment by renchap** [comment]
I agree, this specific error should be handled better. Tagging as confirmed to make it clear that we will accept a PR fixing this.

**Comment by theredled** [comment]
Now leads to a real 404 page, it seems, which fixes part of the issue (still not possible to block the user though)
